### ISSUE TITLE
ALSA incomplete message on Pi

### ISSUE THREAD

**Issue by peladonian** [SpotlightKid/python-rtmidi#179]
I am getting the "UserWarning: MidiOutAlsa::sendMessage: incomplete message!" when sending MIDI from a Raspberry Pi.

Here is an example string that causes the error:
"f0,43,30,3e,19,b0,62,31,b0,63,26,b0,06,7f,b0,26,7f,f7"

Transmitting this from Windows works fine, but fails when I run it from the Pi. I can send it from the Pi using "amidi -S".

**Comment by SpotlightKid** [comment]
> Here is an example string that causes the error:
> "f0,43,30,3e,19,b0,62,31,b0,63,26,b0,06,7f,b0,26,7f,f7"

`b0` (`0xB0`) is not a valid byte in a SysEx message. All data bytes between 0xFO (start of exclusive) and 0xF7 (end of exclusive) must *not* have the high bit set.

**Comment by peladonian** [comment]
It's not a SysEx message, it's NRPN data to a Yamaha mixing console. This particular command turns on the EQ for a specific channel. If I send it from my Windows laptop it is transmitted and the console responds correctly. If I send it from the Pi, ALSA tells me it's incomplete.

**Comment by SpotlightKid** [comment]
If its starts with 0xF0, it is a SysEx message per the spec.

http://www.somascape.org/midi/tech/spec.html#sysexmsgs

If you can show me the manual with the MIDI implementation of the QL5, maybe I can tell you, what the correct message would be. Maybe the data bits need to split into 4-bit nibbles?

**Comment by peladonian** [comment]
I was able to spend some time last night after all. The QL doesn't seem to care about 0x30 vs 0x10, both worked the same for me, despite "echo" being off. It worked without the first three 0xb0's, but stopped when I took out the last one. I get the feeling that there's probably a way to "word" my request differently that would be compliant, so I'm going to conclude that it isn't a python-rtmidi issue. Thanks for your help!
